lib/cocoapods/installer/analyzer.rb

```diff
@@ -429,17 +429,28 @@ def generate_target(target_definition, pod_targets)
           end
         end
 
-        target.pod_targets = pod_targets.select do |pod_target|
-          target_definition_included = pod_target.target_definitions.include?(target_definition)
-          explicitly_defined_in_target_definition = target_definition.non_inherited_dependencies.map(&:name).include?(pod_target.name)
-          next false unless target_definition_included
-          next true if explicitly_defined_in_target_definition
-          !pod_target_test_only?(pod_target, pod_targets)
-        end
+        target.pod_targets = filter_pod_targets_for_target_definition(pod_targets, target_definition)
 
         target
       end
 
+      # Returns a filtered list of pod targets that should or should not be part of the target definition. Pod targets
+      # used by tests only are filtered.
+      #
+      # @param [Array<PodTarget>] pod_targets
+      #        the array of pod targets to check against
+      #
+      # @param [TargetDefinition] target_definition
+      #        the target definition to use as the base for filtering
+      #
+      # @return [Array<PodTarget>] the filtered list of pod targets.
+      #
+      def filter_pod_targets_for_target_definition(pod_targets, target_definition)
+        pod_targets.select do |pod_target|
+          pod_target.target_definitions.include?(target_definition) && !pod_target_test_only?(pod_target, pod_targets)
```

Comment:
a pod target should be part of an aggregate target if:

1) It's included in its target definition (as it has always been before test specs)
2) It is not used only by tests.

For example, 'OCMock' IS part of the target definition BUT it is only referenced by only pod targets as their test dependency therefore it should NOT be part of the aggregate target.